R/LoadingSaving.R

```diff
@@ -211,27 +219,64 @@ getAndromedaTempDiskSpace <- function(andromeda = NULL) {
   if (!is.null(andromeda) && !inherits(andromeda, "SQLiteConnection")) 
     stop("Andromeda argument must be of type 'Andromeda'.")
   
-  # Using Java because no cross-platform functions available in R:
-  if (!.isInstalled("rJava")) {
-    return(NA)
+  if (is.null(andromeda)) {
+    folder <- .getAndromedaTempFolder()
   } else {
-    if (is.null(andromeda)) {
-      folder <- .getAndromedaTempFolder()
+    folder <- dirname(andromeda@dbname) 
+  }
+  
+  if(!dir.exists(folder)) {
+    msg <- paste("Andromeda temp folder does not exist\n",
+                 "folder:", folder, "\n",
+                 "Cannot determine available Andromeda disk space.")
+    rlang::warn(msg, class = "Andromeda")
+    return(NA)
+  }
+  
+  osInfo <- tolower(paste(sessionInfo()$platform, sessionInfo()$running))
```

Comment:
I think the more typical way to determine the OS is by using, `.Platform$OS.type` for example like [here](https://github.com/r-dbi/RSQLite/blob/f7ecf8632972fcecf78d35375d736c08032bcace/tests/testthat/test-encoding.R#L37). Any reason you're using this approach here?